collectors/proc.plugin/proc_stat.c

```diff
@@ -557,7 +694,7 @@ int do_proc_stat(int update_every, usec_t dt) {
                 else
                     rrdset_next(st_scaling_cur_freq);
 
-                chart_per_core_files(&all_cpu_charts[1], all_cpu_charts_size - 1, SCALING_CUR_FREQ_INDEX, st_scaling_cur_freq, 1, 1000, RRD_ALGORITHM_ABSOLUTE);
+                chart_per_core_files(&all_cpu_charts[1], all_cpu_charts_size - 1, CPU_FREQ_INDEX, st_scaling_cur_freq, 1, 1000, RRD_ALGORITHM_ABSOLUTE);
```

Comment:
netdata has a nice feature in for calculating things like this:

```
RRD_ALGORITHM_PCENT_OVER_DIFF_TOTAL
```

Using this algorithm, you can pass the incremental values collected as-is (not need to find the delta), and netdata will calculate the incremental sum of all dimensions and then the percentage of each dimension. We use this for all CPU charts that report ticks.

Reply:
We use ticks only for calculating the average frequency in the case of `time_in_state` is available. So, the algorithm is the correct one.